### PR TITLE
master wowl report ges svs

### PR DESCRIPTION
**Pull request by SimonGenin**

### CODE REVIEW

File: addons/web/static/src/webclient/actions/ux_hooks.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+/**
+ * Hook used to enrich html and provide automatic links to action.
+ * Dom elements must have those attrs [res-id][res-model][view-type]
+ * Each element with those attrs will become a link to the specified resource.
+ * Works with Iframes.
+ *
+ * @param {owl reference} owlRef Owl ref to the element to enrich
+ * @param {string} [selectors] Selectors to apply to the element resolved by the ref.
+ */
+export function useEnrichWithActionLinks(owlRef, selectors = null) {
```
Comment (kebeclibre) [inline addons/web/static/src/webclient/actions/ux_hooks.js]: selector (singular)

File: addons/web/static/src/webclient/actions/ux_hooks.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            let nodeFromRef = owlRef.el;
+
+            // If we get an iframe, we need to wait until everything is loaded
+            if (nodeFromRef.matches("iframe")) {
+                nodeFromRef.onload = () => enrich(comp, nodeFromRef, selectors, true);
+            } else {
+                enrich(comp, nodeFromRef, selectors);
+            }
+        },
+        () => [owlRef, selectors]
```
Comment (kebeclibre) [inline addons/web/static/src/webclient/actions/ux_hooks.js]: if selector is a string, putting it in dependencies is useless

File: addons/web/static/src/webclient/actions/ux_hooks.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * Dom elements must have those attrs [res-id][res-model][view-type]
+ * Each element with those attrs will become a link to the specified resource.
+ * Works with Iframes.
+ *
+ * @param {owl reference} owlRef Owl ref to the element to enrich
+ * @param {string} [selectors] Selectors to apply to the element resolved by the ref.
+ */
+export function useEnrichWithActionLinks(owlRef, selectors = null) {
+    const comp = useComponent();
+
```
Comment (kebeclibre) [inline addons/web/static/src/webclient/actions/ux_hooks.js:18]: let tangibleRefEl;

File: addons/web/static/src/webclient/actions/ux_hooks.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    useEffect(
+        () => {
+            let nodeFromRef = owlRef.el;
+
```
Comment (kebeclibre) [inline addons/web/static/src/webclient/actions/ux_hooks.js:23]: tangibleRefEl = owlRef.el;

File: addons/web/static/src/webclient/actions/ux_hooks.js (hunk 2 above)
Comment (kebeclibre) [reply addons/web/static/src/webclient/actions/ux_hooks.js]: dependenicesfunction should become
`() => [tangibleRefEl]`

**Comment by SimonGenin** [comment]
Switch to master and rebased https://github.com/odoo/odoo/pull/72638

**Pull request closed without merging**